### DISCUSSION TITLE
Add `no-hover` class to button

### DISCUSSION THREAD

**Discussion by runyasak** [saadeghi/daisyui#3213] [Ideas / feature requests]
This discussion is related to #3131.

Hi folks, I want to apply the no-hover class to a button to disable the hover effect.
I've encountered situations where I need to disable the hover effect on buttons.

I’ve forked the project and updated the class, which results in the following behavior:
https://github.com/user-attachments/assets/f190eb2f-ab91-4754-b63b-1b95ce2b211f

Here is my code:
https://github.com/runyasak/daisyui/commit/74d512d79d5cbd44889391c6f755ee03bbbf281a

If you’d like this feature in DaisyUI, I would be happy to contribute it through a pull request.

If you have any feedback or suggestions, please let me know.

**Comment by saadeghi** [comment]
Thanks.  
Can you tell me more about your use case and why it was needed to remove the hover style?

**Reply by runyasak** [reply]
Sure. In some of my projects, I need buttons without any animations or hover effects. I managed this by overriding the background color for the hover state of those buttons by myself.

I believe it would be more convenient if DaisyUI provided a utility class to remove the hover effect.

**Reply by saadeghi** [reply]
I see...  
My question is why hover effect is not needed?  
Is it an aesthetic decision? Because having a slight color change is a better UX generally. It helps users to realize it's a clickable element.

**Reply by runyasak** [reply]
Some of the clients requested to disable the hover effect after seeing it.
I agree that the hover effect improves UX, but I have to disable it for them due to client requirements.
I don’t consider this feature essential, but having an optional way to address cases like this would be helpful.

**Comment by aa365** [comment]
it would be useful for bootstrap style [toggle buttons](https://getbootstrap.com/docs/5.3/forms/checks-radios/#outlined-styles). It's possible to get something similar using not-has-checked:

`<label class="btn btn-primary not-has-checked:btn-outline" for="item_11"><input type="checkbox" value="11" name="item_id" id="item_11">Some item</label>`

But after unchecking an input, the solid style persists because of the hover effect
